### PR TITLE
(PXP-7219) Update Fence to use latest gen3authz and cdiserrors 

### PR DESCRIPTION
**Pull request by BinamB**
The newer version of gen3authz introduced some async components to it which the usersync cant handle at the moment so created a new endpoint in gen3authz so that usersync doesn't have to interact with any async code. [gen3authz#33](https://github.com/uc-cdis/gen3authz/pull/33)

### Bug Fixes
- Fixed usersync not working due to async components of newer version of gen3authz. 
- Fixed some tests not working with newer version of gen3authz. 

### Dependency updates
- Updated `authutils=^5.0.5`
- Updated `cdispyutils=^1.0.5`
- Updated `gen3authz=^1.0.4`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -13,21 +13,21 @@ include = [
...
-cdispyutils = "^1.0.0"
+cdispyutils = { git = "https://github.com/uc-cdis/cdis-python-utils.git", branch = "chore/cdiserrors-dep"}
 cryptography = "==2.8"
 flask = "^1.1.1"
 flask-cors = "^3.0.3"
 flask-restful = "^0.3.6"
 flask_sqlalchemy_session = "^1.1"
 email_validator = "^1.1.1"
-gen3authz = "^0.4.0"
+gen3authz = { path = "./pkg/gen3authz-1.0.34.tar.gz"}
```
Comment (paulineribeyre) [inline pyproject.toml:30]: cool, you should write a msg in the arcadia channel to explain how to test this thing, for future reference :D

File: fence/sync/sync_users.py [hunk 2]
```diff
@@ -1502,9 +1502,7 @@ def _update_authz_in_arborist(self, session, user_projects, user_yaml=None):
         # from authorization sources get policies revoked
         arborist_user_projects = {}
         try:
-            arborist_users = self.arborist_client.get_users(
-                url=self.arborist_client._user_url
```
Comment (paulineribeyre) [inline fence/sync/sync_users.py:1506]: there's a reference to `_user_url` in the tests that i think we don't need anymore

**Review by themarcelor — APPROVED** [review]
+1

**Review by paulineribeyre — APPROVED** [review]
nice!!!

**Pull request merged**